### PR TITLE
HTK Utility updates for sharing services

### PR DESCRIPTION
**Pull request by StephenHodgson**
TL;DR-Cleans up the project for smoother deployment as a fully functional asset package w/sharing services

Resolves https://github.com/Microsoft/HoloToolkit-Unity/issues/490
Resolves https://github.com/Microsoft/HoloToolkit-Unity/issues/487
Resolves https://github.com/Microsoft/HoloToolkit-Unity/issues/374

- Fixes issue with unreadable dark text in config windows with Editor Pro Skin enabled.
- Adds options for Sharing Service configuration in HoloLens Project Settings Config Window.
capture
- Adds options for Sharing Service capabilities in HoloLens Capabilities Config Window.
capture
- Adds a option to manually update External Resources via:
 ```HoloToolkit -> Configure -> Update External Resources ```
capture
- Removed duplicated text serialization option.
- Added a duplicate license file at the root of the HoloToolkit so it's included in asset packages.
- Updated Sharing Menu to notify users to enable sharing services if executable cannot be found.

---
The main goal of the bundled external resources is to ensure when people import the asset package they have everything they need to get started **all in one place**.  Currently you'd have to go and find the sharing services, download them, then make sure they're in the correct place yourself.

My thoughts that influenced this change:
1. This project is a simple framework for learning & sharing with the community.
2. This project is for staging and deploying an asset package that people will download, either from here or the Unity Asset Store, and should contain all the resources needed.
3. For advanced developers to download and tailor this project  by exporting an asset package that fits their needs.

Currently if someone was to create a new project and import the current asset package, and try to use any of the menu items related to the sharing service they will get errors:
image
The second message talking about manually coping some .exe is rather vague. It doesn't even give you a clue where to obtain it, considering that external folder doesn't exist if you've only downloaded just the asset package and not the whole HTK project.

### CODE REVIEW

**Comment by riverar** [comment]
Seems like a lot of unrelated changes bunched together, not appropriate for a single PR.

File: Assets/HoloToolkit/Utilities/Plugins/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+This folder is for the Externals.zip that should alway be up to date.
```
Comment (NeerajW) [inline Assets/HoloToolkit/Utilities/Plugins/README.md]: Why are we doing this? Just because we are unable to add the externals folder into the asset package?
One of the main reasons for making it external was they are binary files that don't compile under Assets.
Adding a zip file with the same stuff does not help that cause.

I think we can come up with a better solution than a zip file under Assets. 
Just for solving the problem of mismatching SharingServer.exe binaries problem, we could always add a check in code in the Launch Sharing Service option to check version numbers. I think that would better than the zip.

What do you think?

Comment (NeerajW) [inline Assets/HoloToolkit/Utilities/Plugins/README.md]: Perhaps print a message, please download latest sharing binaries or something akin.

Comment (S-Hodgson-MMOARgames) [inline Assets/HoloToolkit/Utilities/Plugins/README.md]: This was just a placeholder file I meant to delete.

But the purpose of the zip file is so that users have all the assets they need when they import the asset package the first time. The external binaries are in the zip file and are only unpacked when users opt to use the sharing services in the HoloToolkit/Configure/Apply Project Settings menu.

Comment (NeerajW) [inline Assets/HoloToolkit/Utilities/Plugins/README.md]: But it does mean there is another copy of this uncompilable code in the repo now right? I guess that's what I'm not sure if the best option. So when someone updates the External folder they also now need to update the zip file.

Comment (S-Hodgson-MMOARgames) [inline Assets/HoloToolkit/Utilities/Plugins/README.md]: Everything would compile fine, lol.  This PR would create a duplicate of the External resources but in a zipped format.  Ideally this Externals folder should only live in the [HoloToolkit](https://github.com/Microsoft/HoloToolkit) and not this repo, and this repo only having the bundled external assets that gets updated automatically (pretty much what https://github.com/Microsoft/HoloToolkit-Unity/issues/279 talks about).  Currently it would have to be done by hand, but to be honest, the main HoloToolkit doesn't change all that often.

I'm thinking about this repo in three ways:

1. This project is for learning and sharing with the community.
2. This project is for staging and creating the asset package that people will download either from here or the Unity Asset Store and should contain all the resources needed.
3. For people to download and tailor this project to export an asset package that fits their needs.

**Comment by HodgsonSDAS** [comment]
I'm going to hold off on this until the Sharing Services are working again.

**Comment by HodgsonSDAS** [comment]
I'm going to resubmit this and break apart the sharing service updates and the update to the utilities stuff.

The new sharing service PR will only contain fixes,
The new utilities PR will only focus on getting the package ready for production.
